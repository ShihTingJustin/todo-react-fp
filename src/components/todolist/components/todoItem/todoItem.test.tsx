import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import TodoItem from './todoItem';

describe('Test Button component', () => {
  const props = {
    todo: {
      id: '63a86c39dbd466d9f6fae7a2',
      title: 'test',
      listId: '63a84fe8ab46917a7747ff8e',
      completed: false,
    },
    onToggle: () => {},
    onDelete: () => {},
    onBlur: () => {},
  };
  const component = <TodoItem {...props} />;

  it('[Render] match snapshot', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('[Render] toBeInTheDocument', () => {
    render(component);
    const element = screen.getByTestId('todoItem');
    expect(element).toBeInTheDocument();
  });
});
