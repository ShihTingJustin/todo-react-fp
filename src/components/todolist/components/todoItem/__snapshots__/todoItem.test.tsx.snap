// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button component [Render] match snapshot 1`] = `
<div
  className="\\"rowgroup\\" undefined"
  data-testid="todoItem"
  tabIndex={0}
>
  <div
    className="content flex items-center py-3 pr-5 pl-6 w-full"
  >
    <div
      className="mr-3"
    >
      <input
        checked={false}
        className="ant-input toggle hidden"
        id="63a86c39dbd466d9f6fae7a2"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        type="checkbox"
        value=""
      />
      <label
        htmlFor="63a86c39dbd466d9f6fae7a2"
        onClick={[Function]}
      >
        <div
          className="w-[1.375rem] h-[1.375rem]"
        >
          <svg
            className="glyph-box"
            fill="#c4c4c6"
            role="presentation"
            version="1.1"
            viewBox="0 0 99.6099853515625 99.6572265625"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              transform="matrix(1 0 0 1 -8.740039062500045 85.05859375)"
            >
              <path
                d="M58.5449 14.5508C85.791 14.5508 108.35-8.05664 108.35-35.2539C108.35-62.5 85.7422-85.0586 58.4961-85.0586C31.2988-85.0586 8.74023-62.5 8.74023-35.2539C8.74023-8.05664 31.3477 14.5508 58.5449 14.5508ZM58.5449 6.25C35.498 6.25 17.0898-12.207 17.0898-35.2539C17.0898-58.3008 35.4492-76.7578 58.4961-76.7578C81.543-76.7578 100-58.3008 100.049-35.2539C100.098-12.207 81.5918 6.25 58.5449 6.25Z"
              />
            </g>
          </svg>
        </div>
      </label>
    </div>
    <div
      className="todo-item w-full"
    >
      <div
        className="flex items-center w-full border-b border-border-gray2"
      >
        <input
          className="ant-input ant-input-borderless edit w-full text-content-1"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          type="text"
          value="test"
        />
        <div
          className="delete-button px-3 py-1 cursor-pointer"
          onClick={[Function]}
        >
          <span
            aria-label="close"
            className="anticon anticon-close"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="close"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
